lib/internal/streams/duplex.js

```diff
@@ -41,12 +41,10 @@ const Writable = require('internal/streams/writable');
 ObjectSetPrototypeOf(Duplex.prototype, Readable.prototype);
 ObjectSetPrototypeOf(Duplex, Readable);
 
-{
-  // Allow the keys array to be GC'ed.
-  for (const method of ObjectKeys(Writable.prototype)) {
-    if (!Duplex.prototype[method])
-      Duplex.prototype[method] = Writable.prototype[method];
-  }
+// Allow the keys array to be GC'ed.
+for (const method of ObjectKeys(Writable.prototype)) {
+  if (!Duplex.prototype[method])
+    Duplex.prototype[method] = Writable.prototype[method];
```

Comment:
We should be using a `for(;;)` for performance reasons here. Maybe this can go in a separate commit/PR, as it's not exactly related to the other changes.
```suggestion
{
  const keys = ObjectKeys(Writable.prototype);
  // Allow the keys array to be GC'ed.
  for (let i = 0; i < keys.length; i++) {
    const method = keys[i];
    if (!Duplex.prototype[method])
      Duplex.prototype[method] = Writable.prototype[method];
  }
```